script/approver.go

```diff
@@ -0,0 +1,102 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"path/filepath"
+	"strings"
+)
+
+type Approver struct {
+	gitHub      GitHub
+	application Application
+}
+
+func (a *Approver) Approve() {
+	a.gitHub = GitHub{}
+	a.application = Application{}
+
+	if err := a.application.SetApprover(); err != nil {
+		a.printErrorAndExit(err)
+	}
+
+	if err := a.gitHub.Init(); err != nil {
+		a.printErrorAndExit(err)
+	}
+
+	if *a.gitHub.Issue.State == "closed" {
+		a.printErrorAndExit(fmt.Errorf("script run on closed issue"))
+	}
+
+	if !a.gitHub.IssueHasLabel(LabelStatusApproved) {
+		a.printErrorAndExit(
+			fmt.Errorf("script run on issue that does not have required '%s' label", LabelStatusApproved),
+		)
+	}
+
+	a.application.Parse(a.gitHub.Issue)
+
+	if !a.application.IsValid() {
+		a.printErrorAndExit(
+			fmt.Errorf("script run on issue with invalid application data:\n\n%s", a.renderProblems()),
+		)
+	}
+
+	// The reviewer may remove this label themselves, but
+	// let's double check and remove it if they haven't
+	if a.gitHub.IssueHasLabel(LabelStatusReviewing) {
+		if err := a.gitHub.RemoveIssueLabel(LabelStatusReviewing); err != nil {
+			a.printErrorAndExit(
+				fmt.Errorf("could not remove issue label '%s': %s", LabelStatusReviewing, err.Error()),
+			)
+		}
+	}
+
+	if err := a.gitHub.CommitNewFile(
+		filepath.Join("data", a.application.FileName()),
+		a.application.GetData(),
+		fmt.Sprintf("Added \"%s\" to program", a.application.Project.Name),
+	); err != nil {
+		a.printErrorAndExit(
+			fmt.Errorf("could not create commit: %s", err.Error()),
+		)
+	}
+
+	if err := a.gitHub.CreateIssueComment(a.getApprovalMessage()); err != nil {
+		a.printErrorAndExit(
+			fmt.Errorf("could not create issue comment: %s", err.Error()),
+		)
+	}
+
+	if err := a.gitHub.CloseIssue(); err != nil {
+		a.printErrorAndExit(
+			fmt.Errorf("could not close issue: %s", err.Error()),
+		)
+	}
+}
+
+func (a *Approver) printErrorAndExit(err error) {
+	log.Fatalf("Error approving application: %s\n", err.Error())
+}
+
+func (a *Approver) renderProblems() string {
+	var problemStrings []string
+
+	for _, err := range a.application.Problems {
+		problemStrings = append(problemStrings, fmt.Sprintf("- %s", err.Error()))
+	}
+
+	return strings.Join(problemStrings, "\n")
+}
+
+func (a *Approver) getApprovalMessage() string {
+	return strings.TrimSpace(fmt.Sprintf(`### 🎉 Your application has been approved
+
+Congratulations, @%s has approved your application! A promotion will be applied to your 1Password account shortly.
+
+If you haven't done so already, we highly recommend implementing a [recovery plan for your team](https://support.1password.com/team-recovery-plan/) in case access for a particular contributor is ever lost. You may add additional core contributors as you see fit.
```

Comment:
[[style]](https://support.apple.com/en-gb/guide/applestyleguide/apsg3acde405/1.0/web/1.0#apd1d1eb8bec58f4) Using curly apostrophes over straight apostrophes

```suggestion
If you haven’t done so already, we highly recommend implementing a [recovery plan for your team](https://support.1password.com/team-recovery-plan/) in case access for a particular contributor is ever lost. You may add additional core contributors as you see fit.
```

There's something about the tone that doesn’t fit well with me. The "highly" or even the "we" and "recommend" all feel out of place. When b5web links out to this `team-recovery-plan`, it uses sentences like:

> To lower the risk of lockout, <link>assign at least one other person to help with account recovery</link>.

> 1Password can't recover your account for you. Talk to your {{ if .Account.IsFamily }}Family Organizer{{ else }}administrator{{ end }} about implementing a Recovery Plan to make sure you never lose access to your items.

> As the only person with permission to recover accounts, your team will rely on you for account recovery. If you forget your account details or misplace your Emergency Kit, you may lose access to your account. To lower risk of lockout, <link>add another person to help with account recovery</link>.

Is there something with a more neutral tone (that's still warm and informal), that [doesn't use “we”](https://support.1password.com/style-guide/#first-person-plural-pronouns) that we could use here? (Effective writing is hard!) Maybe something like:

```suggestion
To lower the risk of lockout, [assign at least one other core contributor to help with account recovery](https://support.1password.com/team-recovery-plan/) in case access for a particular contributor is ever lost.
```

I also realized in writing that sentence that I don't know if you meant "You may add additional core contributors [as members in your 1Password account]" or if you meant "You may add additional core contributors [as recovery group members]". Either way, I like that and I would keep that in there (maybe with the extra information).

Reply:
This is a good point you raise. This language has been around since the [start](https://github.com/1Password/1password-teams-open-source/pull/116#issuecomment-476374404) of the program, but I think I'd like to move away from the "contributor" phrasing since an open source team can be made up of individuals outside that particular title.

> To lower the risk of lockout, [assign at least one other person to help with account recovery](https://support.1password.com/team-recovery-plan/) in case access for a particular team member is ever lost. You may add additional team members to your 1Password account, including for account recovery, as you see fit.